packages/tpetra/core/src/Tpetra_MultiVector_def.hpp

```diff
@@ -329,9 +329,23 @@ namespace { // (anonymous)
     }
   }
 
+  template <class impl_scalar_type, class buffer_device_type>
+  bool
+  runKernelOnHost ( Kokkos::DualView<impl_scalar_type*, buffer_device_type> imports )
```

Comment:
Allowing the user to choose the localLengthThreshold maybe using a parameter list would be better for performance. It is not obvious that all GPUs will have the same threshold.
Also allowing to restore the current behavior using a second parameter seem useful for debugging and for folks who have tests that rely on gold files and such.

Reply:
I agree that some way to control the threshold would be nice, but I don't think there's a reasonable way to do it with a parameter list. MultiVector doesn't get a parameter list provided anywhere right now, and it would be very inconvenient as a user to have to find everywhere we construct MultiVectors and provide a parameter list to change this.

Maybe another function in Tpetra::Details::Behavior that looks for an environment variable? Ultimately this is the kind of thing I'd like the Kokkos auto-tuning feature that is being worked on to control, but I think that's a ways off still.

Reply:
I'm working on making this happen via Tpetra::Details::Behavior right now.